include/swift/AST/DiagnosticsSema.def

```diff
@@ -2988,8 +2988,8 @@ ERROR(invalid_objc_decl,none,
 ERROR(invalid_objc_swift_rooted_class,none,
       "only classes that inherit from NSObject can be declared @objc", ())
 ERROR(invalid_nonobjc_decl,none,
-      "only methods, initializers, properties and subscript declarations can "
-      "be declared @nonobjc", ())
+      "@nonobjc can only be used with members of classes, @objc protocols, and "
```

Comment:
I just didn't think it all the way through. Maybe "@nonobjc can only be used with class members and extensions of classes"?